runtime/near-vm-logic/src/logic.rs

```diff
@@ -943,14 +943,190 @@ impl<'a> VMLogic<'a> {
     pub fn keccak512(&mut self, value_len: u64, value_ptr: u64, register_id: u64) -> Result<()> {
         self.gas_counter.pay_base(keccak512_base)?;
         let value = self.get_vec_from_memory_or_register(value_ptr, value_len)?;
-        self.gas_counter.pay_per_byte(keccak512_byte, value.len() as u64)?;
+        self.gas_counter.pay_per(keccak512_byte, value.len() as u64)?;
 
         use sha3::Digest;
 
         let value_hash = sha3::Keccak512::digest(&value);
         self.internal_write_register(register_id, value_hash.as_slice().to_vec())
     }
 
+    /// Hashes the given value using RIPEMD-160 and returns it into `register_id`.
+    ///
+    /// # Errors
+    ///
+    /// If `value_len + value_ptr` points outside the memory or the registers use more memory than
+    /// the limit with `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + ripemd160_base + ripemd160_byte * num_bytes`
+    #[cfg(feature = "protocol_feature_evm")]
+    pub fn ripemd160(&mut self, value_len: u64, value_ptr: u64, register_id: u64) -> Result<()> {
+        self.gas_counter.pay_base(ripemd160_base)?;
+        let value = self.get_vec_from_memory_or_register(value_ptr, value_len)?;
+        self.gas_counter.pay_per(ripemd160_byte, value.len() as u64)?;
+
+        use ripemd160::Digest;
+
+        let value_hash = ripemd160::Ripemd160::digest(&value);
+        self.internal_write_register(register_id, value_hash.as_slice().to_vec())
+    }
+
+    /// Hashes the given value using BLAKE2b and returns it into `register_id`.
+    ///
+    /// # Errors
+    ///
+    /// If `value_len + value_ptr` points outside the memory or the registers use more memory than
+    /// the limit with `MemoryAccessViolation`.
+    ///
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + blake2b_base + blake2b_byte * num_bytes`
```

Comment:
Should this be updated since it is 
```
self.gas_counter.pay_per(blake2b_block, message_blocks)?; 
self.gas_counter.pay_per(blake2b_round, (rounds as u64) * message_blocks)?
```